src/univers/versions.py

```diff
@@ -343,3 +344,86 @@ def __gt__(self, other):
         if not isinstance(other, self.__class__):
             return NotImplemented
         return gentoo.vercmp(self.value, other.value) > 0
+
+
+@attr.s(frozen=True, order=False, eq=False, hash=True)
+class LegacyOpensslVersion(Version):
+    @classmethod
+    def is_valid(cls, string):
+        return OpensslVersion.is_valid_legacy(string)
+
+    def __eq__(self, other):
+        if isinstance(other, SemverVersion):
+            return False
+        if not isinstance(other, self.__class__):
+            return NotImplemented
+        return self.value == other.value
+
+    def __lt__(self, other):
+        """
+        All legacy scheme is behind those using semver.
+        """
+        if isinstance(other, SemverVersion):
+            return True
+        if not isinstance(other, self.__class__):
+            return NotImplemented
+        re_vers1 = re.findall(r"^(\d+\.\d+\.\d+)(.*)$", self.value)[0]
+        re_vers2 = re.findall(r"^(\d+\.\d+\.\d+)(.*)$", other.value)[0]
+        if int(re_vers1[0].replace(".", "")) < int(re_vers2[0].replace(".", "")):
+            return True
+        return re_vers1[0] == re_vers2[0] and re_vers1[1] < re_vers2[1]
```

Comment:
This is cryptic. Can you make it simpler ?

Reply:
Is this good enough??
```python
regexed_vers_self = re.findall(r"^(\d+\.\d+\.\d+)(.*)$", self.value)[0]
regexed_vers_other = re.findall(r"^(\d+\.\d+\.\d+)(.*)$", other.value)[0]

base_vers_self = regexed_version_self[0]
base_vers_other = regexed_version_other[0]

numeric_base_version_self = int(base_vers_self.replace(".","")
numeric_base_version_other = int(base_vers_other.replace(".","")

postfix_self = regexed_version_self[1]
postfix_other = regexed_version_other[1]

if numeric_base_version_self < numeric_base_version_other:
      return True
return base_vers_self == base_vers_other and postfix_self < postfix_other

```